apps/sync/js/sync-engine/syncengine.js

```diff
@@ -0,0 +1,358 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * SyncEngine
+ *
+ * The SyncEngine constructor takes care of:
+ * - Setting up the kinto.js client.
+ * - Configuring kinto.js to connect to the remote Syncto server (which in turn
+ *   proxies all requests to the remote Firefox Sync server).
+ * - Creating kinto.js collections for the 'meta' and 'crypto' collections, with
+ *   the appropriate IdSchemas (the 'meta' collection only contains one record,
+ *   called 'global', and the 'crypto' collection only contains one record,
+ *   called 'keys').
+ * - Setting up the FxSyncWebCrypto instance to be used for cryptographically
+ *   encoding / decoding records, before uploading / after downloading them.
+ * - Registering the DataAdapters to be used on each collection (see below).
+ *
+ * The `syncNow` method takes care of:
+ * - If called for the first time, or if this failed the previous time:
```

Comment:
The user will be an option in the SyncEngine constructor (see bug	1202627), so it cannot change for a given SyncEngine instance.